runtime/runtime.m

```diff
@@ -270,8 +270,10 @@
 	if (index == -1) {
 		p = mono_signature_get_return_type (msig);
 	} else {
-		for (int i = 0; i < index + 1; i++)
+		for (int i = 0; i < index + 1; i++) {
+			xamarin_mono_object_release (&p);
 			p = mono_signature_get_params (msig, &iter);
+		}
```

Comment:
that seems duplicated inside https://github.com/xamarin/xamarin-macios/pull/11730

same comments
* first iteration p == NULL, is that fine ?
* last iteration, should not `p` be released ?

Reply:
both answered in https://github.com/xamarin/xamarin-macios/pull/11730

Reply:
yeah, I repeated myself...

Hopefully there won't be any merge conflicts!